### PR TITLE
Add Kafka Java client examples for KoP with token authentication

### PR DESCRIPTION
**Pull request by BewareMyPower**
NOTE: The token authentication in KoP is just the PLAIN mechanism.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
@tuteng I think we can add client of other languages like C/C++/Golang/PHP/Python/NodeJS later. It might take some time to add another language's client and related document.

**Comment by BewareMyPower** [comment]
@tuteng PTAL

**Review by tuteng — APPROVED** [review]

**Comment by tuteng** [comment]
> @tuteng I think we can add client of other languages like C/C++/Golang/PHP/Python/NodeJS later. It might take some time to add another language's client and related document.

Agree +1

**Review by Jennifer88huang — APPROVED** [review]

File: kop/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Overview
+
+This directory includes examples of how Kafka clients connect to KoP cluster through Token authentication plugin.
```
Comment (Huanli-Meng) [inline kop/README.md:3]:
```suggestion
This directory includes examples of how Kafka clients connect to a KoP cluster through the Token authentication plugin.
```

File: kop/java/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# Overview
+
+This document describes how to produce messages to and consume messages from a KoP cluster using the Kafka client.
+
+# Prerequisites
+
+- Java 1.8 or higher version
+- Maven
+
```
Comment (Huanli-Meng) [inline kop/java/README.md:9]: > **Note**

File: kop/java/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+> Currently it uses Kafka client 2.0.0. If you want to use another version of Kafka client, you can change the `kafka.version` property in `pom.xml`. The supported version is from 1.0.0 to 2.6.0.
```
Comment (Huanli-Meng) [inline kop/java/README.md:10]:
```suggestion
> This example uses Kafka client 2.0.0. If you want to use another version of Kafka client, you can change the `kafka.version` property in `pom.xml` file. Kafka client version 1.0.0 - 2.6.0 are supported.
```

File: kop/java/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+# Example
+
+See [KoP Security](https://github.com/streamnative/kop/blob/master/docs/security.md) for how to configure KoP with token authentication. Assuming the target topic name is `my-topic` whose namespace is `public/default`.
```
Comment (Huanli-Meng) [inline kop/java/README.md:14]:
```suggestion
See [KoP Security](https://github.com/streamnative/kop/blob/master/docs/security.md) for how to configure KoP with token authentication. This example takes a topic named `my-topic` under `public/default` namespace as reference.
```

File: kop/java/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+1. Grant produce and consume permissions to the specific role.
+
+   ```bash
+   bin/pulsar-admin namespaces grant-permission public/default \
+     --role <role> \
+     --actions produce,consume
+   ```
+
+   NOTE: the `conf/client.conf` should be configured, see [Configure CLI Tools](http://pulsar.apache.org/docs/en/security-jwt/#cli-tools).
```
Comment (Huanli-Meng) [inline kop/java/README.md:24]:
```suggestion
   > **Note**
   >
   > The `conf/client.conf` should be configured. For details, see [Configure CLI Tools].(http://pulsar.apache.org/docs/en/security-jwt/#cli-tools).
```

File: kop/java/src/main/resources/token.properties [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+bootstrap.servers=localhost:9092
+topic=persistent://public/default/my-topic
+group=my-group
+namespace=public/default
+token=token:eyJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJyb290In0.gt-c0gy2JL6LyXziRfW3DbiTxsaWDaQEFuWRiYkn2Qt5mmbzHWPZw2LI06YcK3n2ek\
```
Comment (Huanli-Meng) [inline kop/java/src/main/resources/token.properties:19]: should we use a fake token here?

**Review by Huanli-Meng — COMMENTED** [review]
leave my comments. PTAL

**Comment by BewareMyPower** [comment]
PTAL again @Huanli-Meng

**Review by Huanli-Meng — APPROVED** [review]
LGTM

**Comment by codelipenghui** [comment]
@tuteng @Huanli-Meng Could you please help review this PR?

**Comment by tuteng** [comment]
LGTM

**Pull request merged**